### PR TITLE
[image_picker] add resize quality

### PR DESCRIPTION
**Pull request by katout**
The image size is very large in the current implementation.
I added items that can set jpeg's quality

e.g.

```
ImagePicker.pickImage(
  source: ImageSource.camera,
  maxWidth: maxWidth,
  maxHeight: maxHeight,
  resizeQuality: resizeQuality,
)
```

### CODE REVIEW

**Comment by katout** [comment]
ok!

**Comment by nosmirck** [comment]
hello @katout 

I'm about to make a PR on your branch, please take a look, I didn't change anything really, mostly got rid of the unnecessary changes (gradle and other android files) plus one minor change in the ImageResizer to get the quality before passing it to the method and another in the dart file to validate that the resizeQuality passed is between 0.0 and 1.0

File: packages/image_picker/android/build.gradle [hunk 1]
```diff
@@ -5,17 +5,19 @@ buildscript {
     repositories {
         google()
         jcenter()
+        mavenLocal()
```
Comment (nosmirck) [inline packages/image_picker/android/build.gradle]: unnecessary change

File: packages/image_picker/android/build.gradle [hunk 2]
```diff
@@ -5,17 +5,19 @@ buildscript {
...
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath 'com.android.tools.build:gradle:3.2.0'
```
Comment (nosmirck) [inline packages/image_picker/android/build.gradle]: unnecessary change

File: packages/image_picker/android/build.gradle [hunk 3]
```diff
@@ -5,17 +5,19 @@ buildscript {
...
     }
 }
 
 rootProject.allprojects {
     repositories {
         google()
         jcenter()
+        mavenLocal()
```
Comment (nosmirck) [inline packages/image_picker/android/build.gradle]: unnecessary change

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java [hunk 4]
```diff
@@ -58,6 +60,8 @@ private File resizedImage(String path, Double maxWidth, Double maxHeight) throws
     boolean shouldDownscaleHeight = hasMaxHeight && maxHeight < originalHeight;
     boolean shouldDownscale = shouldDownscaleWidth || shouldDownscaleHeight;
 
+    resizeQuality = resizeQuality != null ? resizeQuality : new Double(1.0);
```
Comment (nosmirck) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java]: int quality = resizeQuality != null ? ((int) resizeQuality * 100.0) : 100;

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java [hunk 5]
```diff
@@ -85,7 +89,7 @@ private File resizedImage(String path, Double maxWidth, Double maxHeight) throws
 
     Bitmap scaledBmp = Bitmap.createScaledBitmap(bmp, width.intValue(), height.intValue(), false);
     ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    scaledBmp.compress(Bitmap.CompressFormat.JPEG, 100, outputStream);
+    scaledBmp.compress(Bitmap.CompressFormat.JPEG, (int) (resizeQuality * 100), outputStream);
```
Comment (nosmirck) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java]: scaledBmp.compress(Bitmap.CompressFormat.JPEG, quality, outputStream);

File: packages/image_picker/example/android/build.gradle [hunk 6]
```diff
@@ -2,17 +2,19 @@ buildscript {
     repositories {
         google()
         jcenter()
+        mavenLocal()
```
Comment (nosmirck) [inline packages/image_picker/example/android/build.gradle]: unnecessary change

File: packages/image_picker/example/android/build.gradle [hunk 7]
```diff
@@ -2,17 +2,19 @@ buildscript {
...
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath 'com.android.tools.build:gradle:3.2.0'
```
Comment (nosmirck) [inline packages/image_picker/example/android/build.gradle]: unnecessary change

File: packages/image_picker/example/android/build.gradle [hunk 8]
```diff
@@ -2,17 +2,19 @@ buildscript {
...
     }
 }
 
 allprojects {
     repositories {
         google()
         jcenter()
+        mavenLocal()
```
Comment (nosmirck) [inline packages/image_picker/example/android/build.gradle]: unnecessary change

File: packages/image_picker/example/android/gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip
```
Comment (nosmirck) [inline packages/image_picker/example/android/gradle/wrapper/gradle-wrapper.properties]: unnecessary change

File: packages/image_picker/lib/image_picker.dart [hunk 10]
```diff
@@ -33,6 +33,7 @@ class ImagePicker {
     @required ImageSource source,
     double maxWidth,
     double maxHeight,
+    double resizeQuality,
   }) async {
     assert(source != null);
```
Comment (nosmirck) [inline packages/image_picker/lib/image_picker.dart]: check for value between 0.0 and 1.0
```
if (resizeQuality != null && (resizeQuality < 0 || resizeQuality > 1.0)) {
  throw ArgumentError.value(resizeQuality, 'resizeQuality must be between 0.0 and 1.0');
}
```

**Comment by katout** [comment]
Hi @nosmirck !
Thank you for checking 👍 

> check for value between 0.0 and 1.0

I thought that libraries of android and ios would check that, but I think so too.

> unnecessary change

Certainly, it should be fixed with another PR rather than this PR.
I compiled it into one for the local test to be successful...
I would like the maintainer to decide whether it is acceptable.

**Comment by katout** [comment]
If @nosmirck  threw a pull request to katout, please create another branch from https: //github.com/katout/plugins/tree/add_quality (different from add_quality )and request katout / plugins.

I am sorry, git and English are not good 🤕

**Comment by nosmirck** [comment]
> If @nosmirck threw a pull request to katout, please create another branch from https: //github.com/katout/plugins/tree/add_quality (different from add_quality )and request katout / plugins.
> 
> I am sorry, git and English are not good 🤕

I tried creating a PR to your repo, but apparently your repo is not accepting PRs (or maybe I did something wrong).

You might as well try to pull my branch, delete your image_picker folder inside your branch and replace with mine (is the same thing plus the fixes I added).

Also, if you give me access to your fork, I might just push my changes too.

**Comment by katout** [comment]
I think that it is because you are running fork with flutter/plugins instead of katout/plugins.
and I see Your branch (nosmirck/plugins) history derived, not a derived branch from katout/plugins.
Sorry if different :cry: 

Please do your best because the modifications should remain in the history

**Comment by nosmirck** [comment]
@katout I managed to do it! I had to delete my old fork and fork again from yours, branch from your branch and add my changes.

The PR is there and awaiting :)

It has been a pleasure working with you XD

**Comment by katout** [comment]
Sorry,  I have no motivation to maintain it because it is too old.
Please close this issue as it looks like a little while reading the code.

**Comment by cyanglaz** [comment]
@katout Thank you for the quick response. I do think what the PR is trying to do is valuable. It isn't trivial to merge right now. I'm following the [initial PR review policy](https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#the-review-process), I'm labeling it with "backlog" and we will prioritize according to the issue's priority.

The issue is created here https://github.com/flutter/flutter/issues/36072

**Comment by cyanglaz** [comment]
This feature has been added in https://github.com/flutter/plugins/pull/1896
Closing this PR.
